velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -452,6 +452,39 @@ VeloxToSubstraitExprConvertor::toSubstraitExpr(
   return *substraitFieldExpr;
 }
 
+const ::substrait::Expression&
+VeloxToSubstraitExprConvertor::toSubstraitSingularOrList(
+    google::protobuf::Arena& arena,
+    const std::vector<core::TypedExprPtr>& inputs,
+    const RowTypePtr& inputType,
+    ::substrait::Expression* substraitExpr) {
+  ::substrait::Expression_SingularOrList* singularOrList =
+      substraitExpr->mutable_singular_or_list();
+  VELOX_CHECK(inputs.size() == 2, "Two arguments expected for IN.");
+  singularOrList->mutable_value()->MergeFrom(
+      toSubstraitExpr(arena, inputs.at(0), inputType));
+  auto constantExpr =
+      std::dynamic_pointer_cast<const core::ConstantTypedExpr>(inputs.at(1));
+  VELOX_CHECK_NOT_NULL(constantExpr);
+  VELOX_CHECK(
+      constantExpr->hasValueVector(),
+      "Value vector expected for constant expression.");
```

Comment:
Should we also check the type of the expression? Should it be ARRAY(BIGINT())?

Reply:
Added type check for ARRAY, but the element type can be of other type except for BIGINT.